src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -107,6 +111,14 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
+    await this.startConnectingInternal(reconnecting);
+    return await new Promise<void>((resolve, reject) => {
+      this.bootstrapResolved = resolve;
```

Comment:
Think you can just store the promise directly and then use .resolve or .reject on it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject

Reply:
Tried with the approach you mentioned and faced some issues before I went with this approach recommended in SO.